AayushMathur7/raja-app
# Fix get_repo_info() function to handle trailing slash in repository URLs

PR Body:

**Problem:**
The current implementation of the `get_repo_info()` function in our application is not robust enough to handle different formats of the repository URLs. When the URL ends with a trailing slash, the function fails to return the correct owner and repository name.

**Solution:**
To address this issue, I have made the following changes to the `get_repo_info()` function:

```python
def get_repo_info(url):
    # parse the url and split the path
    parsed_url = urlparse(url.rstrip("/"))  # Remove trailing slash if present
    path_parts = parsed_url.path.split("/")

    # the repo name is the last part of the path
    repo_name = path_parts[-1]

    # the owner is the second-to-last part of the path
    owner = path_parts[-2]

    return owner, repo_name
```

**How to Reproduce:**
1. With the current implementation of the `get_repo_info()` function, call it with a GitHub repository URL that ends with a trailing slash, such as "https://github.com/aayushmathur7/raja-app/".
2. Notice that the function does not return the correct owner and repository name. Instead, it considers the trailing slash as the last part of the path, leading to incorrect results.
3. Call the function with a URL that doesn't end with a trailing slash, such as "https://github.com/aayushmathur7/raja-app".
4. Notice that the function works as expected in this case.

**Acceptance Criteria:**
- The `get_repo_info()` function should correctly parse GitHub repository URLs regardless of whether they end with a trailing slash or not.
- The function should return the correct owner and repository name for the input URL.
- The function should include proper error handling to deal with invalid or incorrectly formatted URLs.
